source/isaaclab/isaaclab/actuators/actuator_pd.py

```diff
@@ -275,15 +293,21 @@ def compute(
 
     def _clip_effort(self, effort: torch.Tensor) -> torch.Tensor:
         # compute torque limits
+        torque_speed_top = self._saturation_effort * (1.0 - self._joint_vel / self.velocity_limit)
+        torque_speed_bottom = self._saturation_effort * (-1.0 - self._joint_vel / self.velocity_limit)
         # -- max limit
-        max_effort = self._saturation_effort * (1.0 - self._joint_vel / self.velocity_limit)
-        max_effort = torch.clip(max_effort, min=self._zeros_effort, max=self.effort_limit)
+        max_effort = torch.clip(torque_speed_top, max=self.effort_limit)
         # -- min limit
-        min_effort = self._saturation_effort * (-1.0 - self._joint_vel / self.velocity_limit)
-        min_effort = torch.clip(min_effort, min=-self.effort_limit, max=self._zeros_effort)
+        min_effort = torch.clip(torque_speed_bottom, min=-self.effort_limit)
 
         # clip the torques based on the motor limits
-        return torch.clip(effort, min=min_effort, max=max_effort)
+        clamped = torch.clip(effort, min=min_effort, max=max_effort)
+        gt_vel_at_effort_lim = self._joint_vel > self._vel_at_effort_lim
+        lt_vel_at_neg_effort_lim = self._joint_vel < -self._vel_at_effort_lim
+        clamped[gt_vel_at_effort_lim] = torque_speed_top[gt_vel_at_effort_lim]
+        clamped[lt_vel_at_neg_effort_lim] = torque_speed_bottom[lt_vel_at_neg_effort_lim]
```

Comment:
I don't think we need to do recomputation here since (torque_speed_bottom, torque_speed_top) shouldn't lie outside (-effort_limit, effort_limit) for cases where velocity is higher than the actual velocity limits.

Maybe I misunderstand a bit, but shouldn't just the clipping once be sufficient?